diff --git i/Makefile w/Makefile
index 6dbda98..aaf550a 100644
--- i/Makefile
+++ w/Makefile
@@ -16,8 +16,8 @@ hello.asm: hello.b b
 b: b.rs libc.rs crust.rs nob.rs stb_c_lexer.rs nob.o stb_c_lexer.o flag.o
 	rustc --edition 2021 -g -C opt-level=z -C link-args="-lc nob.o stb_c_lexer.o flag.o" -C panic="abort" b.rs
 
-nob.o: nob.h
-	clang -g -x c -DNOB_IMPLEMENTATION -c nob.h
+nob.o:
+	clang -g -x c -DNOB_IMPLEMENTATION -o nob.o -c @NOB_H@
 
 stb_c_lexer.o: stb_c_lexer.h
 	clang -g -x c -DSTB_C_LEXER_IMPLEMENTATION -c stb_c_lexer.h
